### PR TITLE
feat(cli): add modern Python dependency manager support to langgraph dev (#5463)

### PR DESCRIPTION
**Pull request by alikhan126**
**Description:** 
This PR adds comprehensive support for modern Python dependency managers (uv, poetry, pipenv) to the `langgraph dev` command. The implementation automatically detects the dependency manager used in a project, activates the appropriate virtual environment, and optionally installs dependencies using the `--install-deps` flag.

**Key Features:**
- Automatic detection of `uv.lock`, `pyproject.toml`, and `Pipfile`
- Virtual environment activation with proper Python path management
- Cross-platform support (Windows/Unix)
- Priority order: uv > poetry > pipenv > pip (uv prioritized as LangGraph's chosen tool)
- Robust error handling and fallback mechanisms
- Working directory preservation during setup
- Dynamic Python version detection for virtual environments
- Optional dependency installation with `--install-deps` flag

**Implementation Details:**
- New `dependency_manager.py` module with comprehensive detection logic
- Integration into `cli.py` dev command with user feedback messages
- 22 unit tests covering all scenarios and edge cases
- Complete example projects for uv, poetry, and pipenv workflows
- Comprehensive documentation with step-by-step usage examples
- Detailed README updates with detection priority and fallback behavior

**Issue:** #5463

**Dependencies:** None (uses existing subprocess and pathlib modules)

### CODE REVIEW

File: libs/cli/langgraph_cli/dependency_manager.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+import subprocess
+import sys
+from enum import Enum
+from typing import Optional
+
+
+class DependencyManager(Enum):
+    """Supported dependency managers."""
+
+    PIPENV = "pipenv"
```
Comment (eyurtsev) [inline libs/cli/langgraph_cli/dependency_manager.py:19]: I don't think we'd consider supporting anything but `pip` and `uv`

**Comment by eyurtsev** [comment]
cc @jdrogers940

**Comment by eyurtsev** [comment]
Closing this PR for now to keep the discussion in the issue
